CircuitPython_Essentials/CircuitPython_HID_Keyboard/code.py

```diff
@@ -36,29 +31,46 @@
 # led = digitalio.DigitalInOut(board.SCK)
 led.direction = digitalio.Direction.OUTPUT
 
-print("Waiting for key pin...")
-
 while True:
-    # Check each pin
-    for key_pin in key_pin_array:
-        if not key_pin.value:  # Is it grounded?
-            i = key_pin_array.index(key_pin)
-            print("Pin #%d is grounded." % i)
+    # main loops for connection and key presses
+
+    while True:
+        # stay in this loop until a connection is available
+        try:
+            # The keyboard object!
+            keyboard = Keyboard(usb_hid.devices)
+            keyboard_layout = KeyboardLayoutUS(keyboard)  # We're in the US :)
+            break  # successfully connected, leave the connection loop
```

Comment:
you probably want this break in an ``else`` clause of the ``try``

Reply:
Personally I prefer the break at this position as it's clearer. I have moved the KeyboardLaoyutUS code out of try/except, though. I think it can't fail.